featuretools/synthesis/dfs.py

```diff
@@ -337,19 +336,19 @@ def get_feature_data(feature):
 
         if isinstance(feature, AggregationFeature):
             if feature.where:
-                where.append(feature)
+                where.add(feature.primitive.name)
             else:
-                agg.append(feature)
+                agg.add(feature.primitive.name)
         elif isinstance(feature, GroupByTransformFeature):
-            groupby.append(feature)
+            groupby.add(feature.primitive.name)
         elif isinstance(feature, TransformFeature):
-            transform.append(feature)
+            transform.add(feature.primitive.name)
 
         feature_deps = feature.get_dependencies()
         if feature_deps:
             dependencies.extend(feature_deps)
 
-        explored.append(feature.get_name())
+        explored.add(feature.get_name())
```

Comment:
This won't catch the case where there are multiple versions of a primitive set with different parameters but one was unused. It wasn't caught before, and the question of what are unique primitive and feature names is outside the scope of this PR, so we'll revisit this at a later date via #1141